### DISCUSSION TITLE
Bachup Wiederherstellung influxdb v1 funktioniert nicht

### DISCUSSION THREAD

**Discussion by jtrouvain** [buanet/ioBroker.docker#448] [Questions & Answers / Fragen & Antworten]
Ich habe auf meiner Synology eine InfkuxDB V2 als Container installiert und eine Influxdb V1. Bei der Iobroker Container Installation habe ich als package INFLUXDB2-CLI mit angegeben. Die Sicherung der Influxdb2 funktioniert so auch.

Nun möchte ich eine Sicherung der Influxdb V1  von meinem Raspberry mit Backitup in den Container der Influxdb V1 einspielen. Hierbei kommt es zum Fehler: ERROR] [influxDB] - /bin/sh: 1: influxd: not found.

Muss ich neben dem Package influxdb2-cli auch ein influxdb-cli instalieren? Kann ich das nachträglich in der Package Variable angeben oder was ist zu tun? Bin im Moment ratlos. Danke für eure Hilfe

**Comment by Marc-Berg** [comment]
Du kannst immer nur EINS der beiden Pakete installieren, da beide teils die gleichen Namen verwenden. Auch bei einer "normalen" Installation wäre das so.
Wenn du nur **einmalig** ein Restore machen willst, tauschst du die ENV-Variable "influxdb2-cli" einfach durch "influxdb" aus, machst die Wiederherstellung und danach die Änderung wieder rückgängig.

Wenn du dauerhaft beide benötigst (weil du parallel InfluxDB 1.x und 2.x backuppen willst), musst du dir das Binary "influxd" (für die InfluxDB 1.x) z.B. unter "/opt/iobroker/bin" kopieren und dann in Backitup darauf referenzieren. Darauf achten, dass die Rechte der kopierten Datei passen.

**Reply by jtrouvain** [reply]
Es gibt kein Verzeichnis /opt/iobroker/bin sondern nur /opt/iobroker/ und dort liegt ein bin file. Hab die Datei influxd dorthin kopiert. Sollte das auch gehen?

**Reply by Marc-Berg** [reply]
Ja, das mit dem /bin war nur ein Beispiel, um eine saubere Struktur zu haben. Es aber egal, wo es liegt, muss nur von backitup zugreifbar sein.

**Comment by jtrouvain** [comment]
Bekomme [ERROR] [influxDB] - /bin/sh: 1: /opt/iobroker: Permission denied
Berechtigung ist wie folgt geändert:
mode of '/opt/iobroker' changed from 0775 (rwxrwxr-x) to 0777 (rwxrwxrwx)
Datei Rechte:
-rwxrwxrwx 1 iobroker iobroker 66656363 Mär 14 11:42 influxd

Was könnte noch fehlen?

**Reply by Marc-Berg** [reply]
Ich weiß jetzt nicht, warum du die Verzeichnisrechte geändert hast, das Anpassen der Datei influxd hätte gereicht. So wie ich das sehe, rufst du nicht "/opt/iobroker/influxd" auf, sondern "/opt/iobroker". Könnte das sein?

**Comment by jtrouvain** [comment]
Ja, richtig. Ich dachte, dass der Pfad reicht, ohne Datei. Habs geändert un der Fehler ist weg.
Jetzt haben wir aber [ERROR] [influxDB] - 2024/03/15 11:53:56 error updating meta: read /opt/iobroker/backups/influxDBtmp: is a directory

Was die Verzeichnisrechte betrifft, da ich kein Experte bin dachte ich, dass man zum Zugriff auch die Verzeichnisrechte ändern muss. habe es wieder zurück gesetzt.

**Comment by jtrouvain** [comment]
Aber wie es aussieht hat er was geladen. Es sind Werte in der DB. Vielen Dank dafür. Wenn man den Fehler wegen influxdbtemp vernachlässigen kann, gut :-), wenn nicht bin ich für weiteren Support dankbar.

**Comment by jtrouvain** [comment]
Hallo,

ich habe jetzt versucht zu sichern. Dabei gibt es folgenden Fehler:
Command failed: "/opt/iobroker/influxd" backup -portable -database iobroker -host 192.168.178.54:8088 "/opt/iobroker/backups/influxDB_2024_03_18-15_23_57_backupiobroker"
2024/03/18 15:23:57 Invalid metadata blob, ensure the metadata service is running (default port 8088)
backup: invalid metadata received

Ich habe bisher folgendes versucht, aber ohne positives Ergebnis:
- bind-address = :8088 gesetzt
- den Standardport 8086 durch 8088  ersetzt
- Firewall der Synology entsprechend angepasst
Irgendeine Idee? Die Beiträge zu diesem Fehler im Forum bringen mich auch nicht weiter

**Reply by Marc-Berg** [reply]
Da habe ich leider keine Idee, was das noch sein könnte. Ich würde (um den Fehler einzugrenzen) erstmal versuchen, das Backup auf dem Server selbst durchzuführen, auf welchem die InfluxDB 1.x läuft.
